packages/p2p/src/hooks/use-p2p-rendered-adverts.ts

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { isMobile } from '@deriv/shared';
+import { useP2PAdvertList } from '@deriv/hooks';
+import { buy_sell } from 'Constants/buy-sell';
+import { useStores } from 'Stores/index';
+
+type TAdvertList = ReturnType<typeof useP2PAdvertList>['data'];
+
+/**
+ * @name getSearchResults
+ * Filter out adverts based on the search term. Btw, this causes a bug that's in production as well.
+ * If a user doesn't scroll to the bottom before searching, the items that fall outside of the limit won't be part of the search results.
+ *  */
+const getSearchResults = (items: TAdvertList, search_term: string) => {
+    if (search_term) {
+        return items?.filter(item =>
+            item.advertiser_details.name.toLowerCase().includes(search_term.toLowerCase().trim())
+        );
+    }
+};
+
```

Comment:
shouldn't this handle the if(!search_term) case as well? may be return an empty array.
`const getSearchResults = (items: TAdvertList, search_term: string) => {
    if (search_term) {
        return items?.filter(item =>
            item.advertiser_details.name.toLowerCase().includes(search_term.toLowerCase().trim())
        );
    }
    return [];
};`

Reply:
Ideally @nada-deriv, this should also be a separate function that handles filtering for `buy / sell` type, but it wouldn't be feasible in my opinion to create such a small function, just to use it in one place. Adding it to the `getSearchResults` violates the single responsibility rule. This logic can be left as it was in the hook, but due to the many conditionals, the hook has a high cognitive complexity. That's why I separated this out.

Reply:
agreed, we can just have the check be `if(search_results)` then below